server/src/Service/InteractionService.php

```diff
@@ -32,8 +40,22 @@ public function getInteraction(
 		$this->validateUsers( $users );
 		$this->validateLimit( $limit );
 
+		// get user ids from usernames
+		foreach ( $users as $username ) {
```

Comment:
nit-pick... use [array_map()](http://php.net/manual/en/function.array-map.php)

Reply:
I'll change it if you want but what's the benefit? It was my understanding that foreach is more efficient than array_map

Reply:
I mean, imho, it's cleaner since it's just building a new array. 
```
$userIds = array_map( function( $user ) {
...
}, $users );
```
and a `foreach` allows side effects, an `array_map` doesn't.